dash/backend/src/modules/reports/dao/reports.dao.ts

```diff
@@ -531,113 +531,94 @@ export class ReportsDao {
+        imageDetailsQuery.andWhere('hki.cluster_id', clusterId);
+      }
+      if (namespaces) {
+        imageDetailsQuery.whereIn('hki.namespace', namespaces);
+      }
+      imageDetailsQuery.groupBy(['i.id', 'i.name', 'hki.name']);
+
+      const scanResultsQuery = knex.select([
+        "image_details.saved_dates",
+        "isrs.scanner_name",
+        knex.raw('image_details.id as image_id'),
+        "image_details.image",
+        "image_details.running_in_cluster",
+        "isrs.type",
+        "isrs.severity",
+        "isrs.is_fixable",
+        "image_details.namespaces"
+      ])
+        .from(imageDetailsQuery.as('image_details'))
+        .leftJoin(knex.raw('image_scan_results as isr on isr.image_id = image_details.id and isr.is_latest=true'))
+        .innerJoin(knex.raw('image_scan_results_issues as isrs on isrs.image_results_id = isr.id'));
+
+      if (severityLevels) {
+          if (severityLevels.includes(VulnerabilitySeverity.MAJOR)) {
+              /** Major vulnerabilities are stored as 'High' on this database, ensures that the proper values
+               * are being searched for */
+              severityLevels.push(VulnerabilitySeverity.HIGH);
+          }
+        scanResultsQuery.whereIn('isrs.severity', severityLevels);
+      }
+      if (fixAvailable) {
+          scanResultsQuery.andWhere('isrs.is_fixable', fixAvailable);
+      }
+      const endDateAsDate = parseISO(endDate);
+      const validatedEndDate = endDateAsDate.toISOString().split('T')[0];
+      const startDateAsDate = parseISO(startDate);
+      const validatedStartDate = startDateAsDate.toISOString().split('T')[0];
+
+      const listsQuery = knex
+        .select(['scanner_name', 'image_id', 'image', 'running_in_cluster', 'type', 'severity', 'is_fixable', 'namespaces'])
+        .from(scanResultsQuery.as('scan_results'))
+        .whereRaw(knex.raw(`NOT saved_dates @> '{"${validatedStartDate}", "${validatedEndDate}"}'`));
+      const newVulnerabilities= listsQuery.clone().whereRaw(knex.raw(`NOT saved_dates @> '{"${validatedStartDate}"}'`));
+      const fixedVulnerabilities= listsQuery.clone().whereRaw(knex.raw(`NOT saved_dates @> '{"${validatedEndDate}"}'`));
+      const summaryQuery = knex.select([
+        knex.raw(`SUM(CASE WHEN NOT saved_dates @> '{"${validatedEndDate}"}' THEN 1 ELSE 0 END) AS fixed`),
+        knex.raw(`SUM(CASE WHEN NOT saved_dates @> '{"${validatedStartDate}"}' THEN 1 ELSE 0 END) AS new`),
+        knex.raw(`SUM(CASE WHEN saved_dates @> '{"${validatedEndDate}", "${validatedStartDate}"}' THEN 1 ELSE 0 END) AS "persistent"`),
+      ])
+        .from(scanResultsQuery.as('scan_results'));
+
+      return Promise.all([
+        newVulnerabilities.then(response => {
+          return plainToClass(ReportsVulnerabilityExportDto, response);
+        }),
+        fixedVulnerabilities.then(response => { return plainToClass(ReportsVulnerabilityExportDto, response); }),
+        summaryQuery,
+      ]).then((values) => {
```

Comment:
There is a typescript feature you can use here. I don't know what the proper name is, but if you have

(values: [var1, var2, ...]) => {...} as your callback definition, you can reference the elements passed in as the values array by the name. This may make this block easier to read if you use that to give the value meaningful names instead of accessing indices of the array.

Reply:
If you remember how exactly to do that I'll gladly put it in! :)

In the meantime, I added declarations as the first part of the .then

Reply:
Sorry for the confusion

Reply:
Darn! I was hoping there was a way to do it haha